### ISSUE TITLE
Incorrect identification of Lint/UselessAccessModifier

### ISSUE THREAD

**Issue by zubin** [rubocop/rubocop#12651]
The `Lint/UselessAccessModifier` cop incorrectly marks the following as an offence, even though it is not:

```ruby
class Foo < Bar
  private
  delegate :baz, to: :qux
end
```

The reason it's not an offence is because the delegated methods are intentionally private. 

--------

## Expected behavior

Don't raise an offence. 

## Actual behavior

Raises `Lint/UselessAccessModifier` offence. 

## Steps to reproduce the problem

See example above. 

## RuboCop version

```
$ rubocop -V
1.59.0 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.2.2) [arm64-darwin23]
```

**Comment by Earlopain** [comment]
Delegated methods aren't private by default. You need to pass `private: true` to get the desired behaviour. The offense looks correct to me.

**Comment by zubin** [comment]
You are correct, thanks for pointing that out.
